tests/unit/unit-walker.cpp

```diff
@@ -115,6 +174,511 @@ TEST_P(UnitWalkerTests, FunSubstitution)
+      ASSERT_EQ(x.second.second.size(), expected_map[x.first].second.size());
+      for (int i = 0; i < x.second.second.size(); i++){
+        EXPECT_EQ(x.second.second[i], expected_map[x.first].second[i]);
+      }
+    }
+  }
+}
+//TODO add test for UF with not just binary
+
+TEST_P(UnitWalkerTests, FreshVars){
+  //testing repeated values
+  SolverConfiguration sc = GetParam();
+  if (sc.is_logging_solver) {
+  Term x = s->make_symbol("x", bvsort);
+  Term y = s->make_symbol("y", bvsort);
+  Term xpx = s->make_term(BVAdd, x, x);
+  Term ypy = s->make_term(BVAdd, y, y);
+  Term xp2 = s->make_term(BVAdd, x, s->make_term(2, bvsort));
+  Term lmt = s->make_term(BVSub, xp2, xpx);
+  Term rhs = s->make_term(BVAdd, ypy, lmt);
+  Term fullform = s->make_term(Equal, xpx, rhs);
+
+//  map<Term, pair<Term, vector<int>>> expected_map;
+  //y: {1,0,0}
+//  pair <Term, vector<int>> n5;
+//  n5.first = fullform;
+//  vector<int> n5path = {1, 0, 0};
+//  n5.second = n5path;
+//  expected_map[b0] = n5;
+  //2: {1,1,0,1}
+//  pair <Term, vector<int>> n7;
+//  n7.first = fullform;
+//  vector<int> n7path = {1, 1, 0, 1};
+//  n7.second = n7path;
+//  expected_map[b1] = n7;
+  //x+2: {1,1,0}
+//  pair <Term, vector<int>> n6;
+//  n6.first = fullform;
+//  vector<int> n6path = {1, 1, 0};
+//  n6.second = n6path;
+//  expected_map[b2] = n6;
+  //x: {0,0}
+//  pair <Term, vector<int>> n2;
+//  n2.first = fullform;
+//  vector<int> n2path = {0, 0};
+//  n2.second = n2path;
+//  expected_map[x] = n2;
+  //y+y: {1,0}
+//  pair <Term, vector<int>> n4;
+//  n4.first = fullform;
+//  vector<int> n4path = {1, 0};
+//  n4.second = n4path;
+//  expected_map[ypy] = n4;
+  //(y+y)+(x+2)-(x+x): {1}
+//  pair <Term, vector<int>> n3;
+//  n3.first = fullform;
+//  vector<int> n3path = {1};
+//  n3.second = n3path;
+//  expected_map[rhs] = n3;
+  //x+x: {0}
+//  pair <Term, vector<int>> n1;
+//  n1.first = fullform;
+//  vector<int> n1path = {0};
+//  n1.second = n1path;
+//  expected_map[xpx] = n1;
+  //x+2-x+x: {1,1}
+//  pair <Term, vector<int>> n8;
+//  n8.first = fullform;
+//  vector<int> n8path = {1, 1};
+//  n8.second = n8path;
+ // expected_map[lmt] = n8;
+  //fullform: {}
+//  pair <Term, vector<int>> n0;
+//  n0.first = fullform;
+//  vector<int> n0path;
+//  n0.second = n0path;
+//  expected_map[fullform] = n0;
+
+  UnorderedTermPairMap UndMap;
+  FreshTreeWalker ftw(s, false, &UndMap);
+
+  pair <Term, vector<int>> p1;
+  p1 = ftw.visit(fullform);
+//  Term = p1.first;
+
+  printMap(UndMap);
+
+ // for (auto const& x : UndMap){
+   // EXPECT_EQ(x.second.first, expected_map[x.first].first); //testing equivalence of first element in pair: the topmost node
+    //ASSERT_EQ(x.second.second.size(), expected_map[x.first].second.size());
+  //  for (int i = 0; i < x.second.second.size(); i++){
+    //  EXPECT_EQ(x.second.second[i], expected_map[x.first].second[i]);
+   // }
+    //}
+  }
+}
+
+
```

Comment:
Please locally: install all solvers but z3:
- msat you already have
- cvc4: using the contrib script
- boolector: using the contrib script
- yices: follow the instrucitons in the contrib script

and then configure with `--msat --cvc4 --yices2 --btor --debug`.